src/extensions/surveys.ts

```diff
@@ -764,6 +777,10 @@ export const createMultipleQuestionSurvey = (posthog: PostHog, survey: Survey) =
                               ].map((choice) => choice.value)
                     response = selectedChoices
                 }
+                const isQuestionOptional = survey.questions[idx].optional
+                if (isQuestionOptional && response === undefined) {
+                    response = null
+                }
```

Comment:
Don't really like the manual index increment, would be nicer if we incorporate it into the for .. of loop so all tabs automatically get the right index.

Specially because it currently requires special handling like here for when response is undefined and we need to increment idx to get the right value for the next question.

Instead do: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/entries#iterating_with_index_and_element

Reply:
I specifically manually increment the index in order to avoid the `Object.entries()` shenanigans required otherwise, I'm fine with either though so I can update.

Reply:
Almost always prefer that to manually incrementing array indexes in js, handles sparse arrays, and makes sure the indexes are always correct.